From 364a0e7343779ead65cbd1b9dba396b20213f6a8 Mon Sep 17 00:00:00 2001
From: Miao Yu <r65093@freescale.com>
Date: Wed, 11 Nov 2009 11:28:25 +0800
Subject: [PATCH 083/280] ENGR00118274 Add fPIC compile flag in ipu, rng,
 screenlayer, sim and vpu lib

Should add -fPIC compile flag when build .c file to .o file,
otherwise there will be runtime error after enable armv7-a build
flag using codesourcery gcc 4.3.3

Signed-off-by: Miao Yu <r65093@freescale.com>
---
 ipu/Makefile         | 2 +-
 rng/Makefile         | 2 +-
 screenlayer/Makefile | 2 +-
 sim/Makefile         | 2 +-
 vpu/Makefile         | 2 +-
 5 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/ipu/Makefile b/ipu/Makefile
index 6bb55b0..36fefef 100644
--- a/ipu/Makefile
+++ b/ipu/Makefile
@@ -28,7 +28,7 @@ $(LIBNAME).so: $(OBJS)
 ifeq "$(PLATFORM)" ""
 	$(error "Unspecified PLATFORM variable")
 endif
-	$(CC) -D$(PLATFORM) $(INCLUDE) -Wall -O2 -c $^ -o $@
+	$(CC) -D$(PLATFORM) $(INCLUDE) -Wall -O2 -fPIC -c $^ -o $@
 
 .PHONY: clean
 clean:
diff --git a/rng/Makefile b/rng/Makefile
index 42aca60..ee70b12 100644
--- a/rng/Makefile
+++ b/rng/Makefile
@@ -6,7 +6,7 @@ LD=$(CROSS_COMPILE)ld
 INC := $(INCLUDE)
 
 # Add compilation checks
-CFLAGS += -g -Wall -W -Wmissing-prototypes -Wstrict-prototypes -Wdeclaration-after-statement
+CFLAGS += -g -Wall -W -Wmissing-prototypes -Wstrict-prototypes -Wdeclaration-after-statement -fPIC
 
 # In absence of kernel CONFIG flags, set API library to build on
 # 'unknown' platform.
diff --git a/screenlayer/Makefile b/screenlayer/Makefile
index 884668a..ba97127 100644
--- a/screenlayer/Makefile
+++ b/screenlayer/Makefile
@@ -26,7 +26,7 @@ $(LIBNAME).so: $(OBJS)
 	$(CC) -shared -nostartfiles -Wl,-soname,$@ -o $@ $^ -L../ipu -lipu
 
 %.o: %.c
-	$(CC) $(INC) -Wall -O2 -c $^ -o $@
+	$(CC) $(INC) -Wall -O2 -fPIC -c $^ -o $@
 
 .PHONY: clean
 clean:
diff --git a/sim/Makefile b/sim/Makefile
index ae8fbe0..0bcdfe8 100644
--- a/sim/Makefile
+++ b/sim/Makefile
@@ -23,7 +23,7 @@ $(LIBNAME).so: $(OBJS)
 	$(CC) -shared -nostartfiles -Wl,-soname,$@ -o $@ $^ -lpthread
 
 %.o: %.c
-	$(CC) -D$(PLATFORM) $(INCLUDE) -Wall -O2 -c $^ -o $@
+	$(CC) -D$(PLATFORM) $(INCLUDE) -Wall -O2 -fPIC -c $^ -o $@
 
 else
 all install :
diff --git a/vpu/Makefile b/vpu/Makefile
index b066b37..29e138d 100644
--- a/vpu/Makefile
+++ b/vpu/Makefile
@@ -39,7 +39,7 @@ all install :
 endif
 
 %.o: %.c
-	$(CC) -D$(PLATFORM) -Wall -O2 -c $^ -o $@
+	$(CC) -D$(PLATFORM) -Wall -O2 -fPIC -c $^ -o $@
 
 $(LIBNAME).so: $(OBJ)
 	$(CC) -shared -nostartfiles -Wl,-soname,$@ $^ -o $@
-- 
1.8.0

